src/index.ts

```diff
@@ -11,42 +14,37 @@ function transpileToCommonJS(code: string) {
   })
 }
 
-/** jest transformers must export a process function
- * @param {src} code to transform
- * @param {path} the path to the file
- */
-export function process(src: string, path: string) {
-  const { descriptor } = parse(src)
-
-  let bindings: BindingMetadata
-  let scriptSetupResult: SFCScriptBlock
-
-  if (descriptor.script || descriptor.scriptSetup) {
-    scriptSetupResult = compileScript(descriptor)
-    bindings = scriptSetupResult.bindings
-  }
-
-  const template = compileTemplate({
-    filename: descriptor.filename,
-    source: descriptor.template.content,
-    compilerOptions: {
-      bindingMetadata: bindings
+class VueTsJestTransformer extends TsJestTransformer implements Transformer {
+  process(input: string, _filePath: Config.Path, _jestConfig: Config.ProjectConfig, _transformOptions?: TransformOptions): TransformedSource | string {
+    const { descriptor } = parse(input)
+    let bindings: BindingMetadata | undefined
+    let scriptSetupResult: SFCScriptBlock
+    let output: TransformedSource | string = ''
+
+    if (descriptor.script || descriptor.scriptSetup) {
+      scriptSetupResult = compileScript(descriptor)
+      bindings = scriptSetupResult.bindings
+      output = transpileToCommonJS(scriptSetupResult.content)
+      const template = compileTemplate({
+        filename: descriptor.filename,
+        source: descriptor.template!.content,
+        compilerOptions: {
+          bindingMetadata: bindings
+        }
+      })
+      const templateCode = transpileToCommonJS(template.code)
+      // this is fairly dirty
+      if (templateCode.includes('exports.render = render;')) {
+        output += templateCode + ';exports.default = {...exports.default, render};'
+      } else {
+        output += templateCode + ';exports.default = {...exports.default};'
+      }
     }
-  })
 
-  const scriptSetupCode = transpileToCommonJS(scriptSetupResult.content)
-  const templateCode = transpileToCommonJS(template.code)
-
-  let output = scriptSetupCode
-
-  // this is fairly dirty
-  if (templateCode.includes('exports.render = render;')) {
-    output += templateCode + ';exports.default = {...exports.default, render};'
-  } else {
-    output += templateCode + ';exports.default = {...exports.default};'
+    // For js, ts, we can call super.process to ask ts-jest to compile. The rest we ask Vue compiler
+    // return super.process(input, filePath, jestConfig, transformOptions);
```

Comment:
Is this commented out for a specific reason? Should we use this?

Reply:
ya that is the important one. If you want `ts-jest` to compile, then you need to deliver `ts` string into that `super.process`